addons/portal/controllers/portal.py

```diff
@@ -196,6 +194,82 @@ def account(self, redirect=None, **post):
         response.headers['X-Frame-Options'] = 'DENY'
         return response
 
+    @http.route('/portal/attachment/add_data', type='json', auth='public')
```

Comment:
how about we rename it to simply /attachment/add to avoid any confusion?
```suggestion
    @http.route('/portal/attachment/add', type='json', auth='public')
```

Reply:
The name `add_data` was for consistency with what we did in `web_editor` (where `data` is opposed to `url`) but OK to change this one here. Done.